streaming/java/streaming-api/src/main/java/io/ray/streaming/api/context/ClusterStarter.java

```diff
@@ -0,0 +1,127 @@
+      File file = new File(socket);
+      if (file.exists()) {
+        file.delete();
+      }
+    }
+
+    // Delete existing socket files.
+    for (String socket : ImmutableList.of(RAYLET_SOCKET_NAME, PLASMA_STORE_SOCKET_NAME)) {
+      File file = new File(socket);
+      if (file.exists()) {
+        file.delete();
+      }
+    }
+
+    String nodeManagerPort = String.valueOf(NetworkUtil.getUnusedPort());
+
+    // jars in the `ray` wheel doesn't contains test classes, so we add test classes explicitly.
+    // Since mvn test classes contains `test` in path and bazel test classes is located at a jar
+    // with `test` included in the name, we can check classpath `test` to filter out test classes.
+    String classpath = Stream.of(System.getProperty("java.class.path").split(":"))
+        .filter(s -> !s.contains(" ") && s.contains("test"))
+        .collect(Collectors.joining(":"));
+    String workerOptions = new Gson().toJson(ImmutableList.of("-classpath", classpath));
+    Map<String, String> config = new HashMap<>(RayConfig.create().rayletConfigParameters);
+    config.put("num_workers_per_process_java", "1");
+    // Start ray cluster.
+    List<String> startCommand = ImmutableList.of(
+        "ray",
+        "start",
+        "--head",
+        "--redis-port=6379",
+        String.format("--plasma-store-socket-name=%s", PLASMA_STORE_SOCKET_NAME),
+        String.format("--raylet-socket-name=%s", RAYLET_SOCKET_NAME),
+        String.format("--node-manager-port=%s", nodeManagerPort),
+        "--load-code-from-local",
+        "--include-java",
+        "--java-worker-options=" + workerOptions,
+        "--internal-config=" + new Gson().toJson(config)
+    );
+    if (!executeCommand(startCommand, 10)) {
+      throw new RuntimeException("Couldn't start ray cluster.");
+    }
+
+    // Connect to the cluster.
+    System.setProperty("ray.redis.address", "127.0.0.1:6379");
+    System.setProperty("ray.object-store.socket-name", PLASMA_STORE_SOCKET_NAME);
+    System.setProperty("ray.raylet.socket-name", RAYLET_SOCKET_NAME);
+    System.setProperty("ray.raylet.node-manager-port", nodeManagerPort);
+    Ray.init();
+  }
+
+  public static void stopCluster(boolean isCrossLanguage) {
+    // Disconnect to the cluster.
+    Ray.shutdown();
+    System.clearProperty("ray.redis.address");
+    System.clearProperty("ray.object-store.socket-name");
+    System.clearProperty("ray.raylet.socket-name");
+    System.clearProperty("ray.raylet.node-manager-port");
+
+    if (isCrossLanguage) {
+      // Stop ray cluster.
+      final List<String> stopCommand = ImmutableList.of(
+          "ray",
+          "stop"
+      );
+      if (!executeCommand(stopCommand, 10)) {
+        throw new RuntimeException("Couldn't stop ray cluster");
+      }
+    }
+  }
+
+  /**
+   * Execute an external command.
+   *
+   * @return Whether the command succeeded.
+   */
+  private static boolean executeCommand(List<String> command, int waitTimeoutSeconds) {
```

Comment:
it's better to move execute shell command function to utils.
Btw, please exit forcibly if it fails to wait for completion